src/dom-to-image-more.js

```diff
@@ -512,6 +541,18 @@
             );
         }
 
+        function hasOpenShadowRoot(value) {
+            return isElement(value) && value.shadowRoot !== null;
```

Comment:
Weird that on this method we're hitting the property `.shadowRoot` of the element directly, while in line 549, we're calling `getWindow()` on the element and asking if the `ShadowRoot` instanceof matches value... I guess I'm wondering if an element's `shadowRoot` _is_ itself when it's a shadow root... e.g. can we do `return value === value.shadowRoot;`? I'm really weak on this whole area, sorry.

Reply:
Or... looking at line 553, could/should _this_ line be `return isInShadowRoot(value) && value === value.getRootNode();`

Reply:
The name of the function should probably be `isElementHostForOpenShadowRoot`